lib/triBoxOverlap.js

```diff
@@ -0,0 +1,212 @@
+    vMin.x = sign * maxBox.x - v;
+    vMax.x = (-sign) * maxBox.x - v;
+
+    v = vertex.y;
+    sign = (normal.y > 0.0) ? -1.0 : 1.0;
+    vMin.y = sign * maxBox.y - v;
+    vMax.y = (-sign) * maxBox.y - v;
+
+    v = vertex.z;
+    sign = (normal.z > 0.0) ? -1.0 : 1.0;
+    vMin.z = sign * maxBox.z - v;
+    vMax.z = (-sign) * maxBox.z - v;
+
+    // Project each vertex as a vector from the origin onto the plane normal
+    if (Cartesian3.dot(normal, vMin) > 0.0) {
+        return false;
+    }
+    else if (Cartesian3.dot(normal, vMax) >= 0.0) {
+        return true;
+    }
+    return false;
+}
+
+////// X-tests //////
+
+function axisTestX01(a, b, fa, fb, boxHalfSize, v0, v2) {
+    var p0 = a * v0.y - b * v0.z;
+	var p2 = a * v2.y - b * v2.z;
+    var min = p2;
+    var max = p0;
+    if (p0 < p2) {
+        min = p0;
+        max = p2;
+    }
+	var rad = fa * boxHalfSize.y + fb * boxHalfSize.z;
+    return (min > rad || max < -rad);
+}
+
+function axisTestX2(a, b, fa, fb, boxHalfSize, v0, v1) {
+    var p0 = a * v0.y - b * v0.z;
+    var p1 = a * v1.y - b * v1.z;
+    var min = p1;
+    var max = p0;
+    if (p0 < p1) {
+        min = p0;
+        max = p1;
+    }
+    var rad = fa * boxHalfSize.y + fb * boxHalfSize.z;
+    return (min > rad || max < -rad);
+}
+
+////// Y-tests //////
+
+function axisTestY02(a, b, fa, fb, boxHalfSize, v0, v2) {
+    var p0 = -a * v0.x + b * v0.z;
+    var p2 = -a * v2.x + b * v2.z;
+    var min = p2;
+    var max = p0;
+    if (p0 < p2) {
+        min = p0;
+        max = p2;
+    }
+    var rad = fa * boxHalfSize.x + fb * boxHalfSize.z;
+    return (min > rad || max < -rad);
+}
+
+function axisTestY1(a, b, fa, fb, boxHalfSize, v0, v1) {
+    var p0 = -a * v0.x + b * v0.z;
+    var p1 = -a * v1.x + b * v1.z;
+    var min = p1;
+    var max = p0;
+    if (p0 < p1) {
+        min = p0;
+        max = p1;
+    }
+    var rad = fa * boxHalfSize.x + fb * boxHalfSize.z;
+    return (min > rad || max < -rad);
+}
+
+////// Z-tests //////
+
+function axisTestZ12(a, b, fa, fb, boxHalfSize, v1, v2) {
+    var p1 = a * v1.x - b * v1.y;
+	var p2 = a * v2.x - b * v2.y;
+    var min = p1;
+    var max = p2;
+    if (p2 < p1) {
+        min = p2;
+        max = p1;
+    }
+	var rad = fa * boxHalfSize.x + fb * boxHalfSize.y;
+    return (min > rad || max < -rad);
+}
+
+function axisTestZ0(a, b, fa, fb, boxHalfSize, v0, v1) {
+    var p0 = a * v0.x - b * v0.y;
+	var p1 = a * v1.x - b * v1.y;
+    var min = p1;
+    var max = p0;
+    if (p0 < p1) {
+        min = p0;
+        max = p1;
+    }
+	var rad = fa * boxHalfSize.x + fb * boxHalfSize.y;
+    return (min > rad || max < -rad);
+}
+
+var triangleNormal = new Cartesian3();
+
+var edges = [];
+edges.push(new Cartesian3());
+edges.push(new Cartesian3());
+edges.push(new Cartesian3());
+
+var v0 = new Cartesian3();
+var v1 = new Cartesian3();
+var v2 = new Cartesian3();
+
+function triBoxOverlap(boxCenter, boxHalfSize, triangle) {
```

Comment:
Can we make this efficiently use Cesium's [AxisAlignedBoundingBox](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/AxisAlignedBoundingBox.js)?